internal/controller/operatorconfigmap_controller.go

```diff
@@ -410,6 +411,9 @@ func (c *OperatorConfigMapReconciler) reconcileDelegatedCSI() error {
 		if c.AvailableCrds[VolumeGroupSnapshotClassCrdName] {
 			csiOperatorConfig.Spec.DriverSpecDefaults.SnapshotPolicy = csiopv1a1.VolumeGroupSnapshotPolicy
 		}
+		if val, ok := c.operatorConfigMap.Data[csiCtrlPluginHostNetworkKey]; ok && val == strconv.FormatBool(false) {
+			csiOperatorConfig.Spec.DriverSpecDefaults.ControllerPlugin.HostNetwork = ptr.To(false)
+		}
```

Comment:
```suggestion
if val, ok := c.operatorConfigMap.Data[csiCtrlPluginHostNetworkKey]; ok && val == strconv.FormatBool(true) {
			csiOperatorConfig.Spec.DriverSpecDefaults.ControllerPlugin.HostNetwork = ptr.To(true)
}
```

Lets the controller plugin run with pod network by default and set it to true only if required?

Reply:
We want to run the controller plugin with Host Network always and only use pod network in case of internal mode scenario where Storage Cluster is on pod network